src/core/core.cpp

```diff
@@ -105,31 +105,26 @@ void Core::registerCallbacks(Tox* tox)
  * @param settings Settings specific to Core
  * @return nullptr or a Core object ready to start
  */
-ToxCorePtr Core::makeToxCore(const QByteArray& savedata, const ICoreSettings& settings,
-                             IBootstrapListGenerator& bootstrapNodes, ToxCoreErrors* err)
+std::pair<ToxCorePtr, Core::ToxCoreErrors> Core::makeToxCore(const QByteArray& savedata,
+                                                             const ICoreSettings& settings,
+                                                             IBootstrapListGenerator& bootstrapNodes)
 {
     auto* thread = new QThread();
     if (thread == nullptr) {
```

Comment:
When refactoring this, why not just remove the unreachable error conditions instead of making the unreachable content nicer? Alternatively, `nothrow`-new could be used, but not sure it's worth it.

Reply:
Removed :).